Python/ceval.c

```diff
@@ -6520,11 +6520,14 @@ _PyEval_GetFrame(void)
 PyFrameObject *
 PyEval_GetFrame(void)
 {
-    PyThreadState *tstate = _PyThreadState_GET();
-    if (tstate->cframe->current_frame == NULL) {
+    _PyInterpreterFrame *frame = _PyEval_GetFrame();
+    while (frame && _PyFrame_IsIncomplete(frame)) {
+        frame = frame->previous;
+    }
+    if (frame == NULL) {
         return NULL;
     }
-    PyFrameObject *f = _PyFrame_GetFrameObject(tstate->cframe->current_frame);
+    PyFrameObject *f = _PyFrame_GetFrameObject(frame);
     if (f == NULL) {
         PyErr_Clear();
     }
```

Comment:
Would it work to use `PyThreadState_GetFrame()`?  There are a couple slight differences,  but mostly it's the same.

```suggestion
    PyThreadState *tstate = _PyThreadState_GET();
    PyFrameObject *f = PyThreadState_GetFrame(tstate);
```

or just

```c
PyFrameObject *
PyEval_GetFrame(void)
{
    PyThreadState *tstate = _PyThreadState_GET();
    return PyThreadState_GetFrame(tstate);
}
```

Reply:
`PyThreadState_GetFrame` returns a new reference, but `PyEval_GetFrame` returns a borrowed one. We could `Py_XDECREF` the result before returning, but that sort of feels icky to me (the only reason it's safe is because we know how `PyThreadState_GetFrame` works).